app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -41,9 +41,9 @@ def redirect_or_render(auth)
     end
 
     def pre_auth
-      @pre_auth ||= OAuth::PreAuthorization.new(Doorkeeper.configuration,
-                                                server.client_via_uid,
-                                                params)
+      @pre_auth ||= Doorkeeper.configuration.preauth_class.new(Doorkeeper.configuration,
```

Comment:
`Doorkeeper.configuration.preauth_class.new(Doorkeeper.configuration`

This reads like hell. I understand why we want it, but I don't like it as is.